### DISCUSSION TITLE
Consider changing template properties to a collection type

### DISCUSSION THREAD

**Discussion by mark-quickel-zz** [microsoft/AzureTRE#1260] [Ideas]
I'm working with the base template, and the properties / system_properties entities are returned as a flattened list of properties, which makes them more challenging to iterate over without converting to a collection. In my case, within Power Apps. Wondering if there is value in having properties return as follows: 

"properties": [
   {
      "name": "display_name",
      "type": "string",
      "title": "Display Name",
      "description": "",
      "default": "display name value"
   }
],...

**Comment by marrobi** [comment]
Hi @mark-quickel 

Are you meaning the payload returned by `api/workspace-templates/<template_name>`?

This is a JSON Schema compliant document, and believe properties need to be an object. See https://json-schema.org/learn/getting-started-step-by-step.html . We can then use this directly in libraries such as https://jsonforms.io/ .

Maybe we could do with a PowerApps version of this!
